client/src/main/java/io/pravega/client/control/impl/Controller.java

```diff
@@ -127,6 +127,20 @@
      */
     CompletableFuture<Boolean> createStream(final String scope, final String streamName, final StreamConfiguration streamConfig);
 
+    /**
+     * API to create an internal stream. The future completes with true in the case the stream did not
+     * exist when the controller executed the operation. In the case of a re-attempt to create
+     * the same stream, the future completes with false to indicate that the stream existed when
+     * the controller executed the operation.
+     *
+     * @param scope Scope
+     * @param streamName Stream name
+     * @param streamConfig Stream configuration
+     * @return A future which will throw if the operation fails, otherwise returning a boolean to
+     *         indicate that the stream was added because it did not already exist.
+     */
+    CompletableFuture<Boolean> createInternalStream(final String scope, final String streamName, final StreamConfiguration streamConfig);
```

Comment:
I don;t see the need for this method to be added to the interface...

Reply:
The internal call to `LocalController` goes from `ControllerEventProcessor`  via interface `Controller`  only hence it was required to be added. But changed the code and using `LocalController` instance directly to method calls